### PR TITLE
bug fix: may cause redundant change peer request

### LINKED ISSUE #182: code may cause redundant change peer requsest
https://github.com/pingcap-incubator/tinykv/blob/b6d7c442e3c9d1eac7192c3bff8372b16a7ae18b/kv/test_raftstore/scheduler.go#L393-L406
code between 396 and 401 does not make sense, it will always return false unless it is so lucky that the first peer in region is the added peer. Also, if there is no peer in region or only one peer that is added peer, code will reach the unreachable part.

### PR DESCRIPTION
**Pull request by xiongjiwei**
Closes #182

### CODE REVIEW

File: kv/test_raftstore/scheduler.go [hunk 1]
```diff
@@ -395,11 +395,12 @@ func (m *MockSchedulerClient) tryFinished(op *Operator, region *metapb.Region, l
 			for _, p := range region.GetPeers() {
 				if add.peer.GetId() == p.GetId() {
 					add.pending = true
-				} else {
-					// TinyKV rejects AddNode.
+					op.Data = add
```
Comment (Connor1996) [inline kv/test_raftstore/scheduler.go]: nice catch. how about making `op.Data` as a pointer

Comment (xiongjiwei) [reply kv/test_raftstore/scheduler.go]: sure,  I change addPeer op data to pointer

File: kv/test_raftstore/scheduler.go [hunk 2]
```diff
@@ -491,7 +491,7 @@ func (m *MockSchedulerClient) removeRegionLocked(region *metapb.Region) {
 func (m *MockSchedulerClient) AddPeer(regionID uint64, peer *metapb.Peer) {
 	m.scheduleOperator(regionID, &Operator{
 		Type: OperatorTypeAddPeer,
-		Data: OpAddPeer{
+		Data: &OpAddPeer{
```
Comment (Connor1996) [inline kv/test_raftstore/scheduler.go:494]: Please change other operators too

Comment (xiongjiwei) [reply kv/test_raftstore/scheduler.go:494]: sorry forget that, already change them

**Review by Connor1996 — APPROVED** [review]

**Pull request merged**